apps/generate_non_factory_sboms.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import logging
+import os
+import subprocess
+
+from expandvars import expandvars
+
+from apps.compose_apps import ComposeApps
+from helpers import status
+
+
+def main(args):
+    status("Loading docker-compose files to find non-factory containers")
+    factory_prefix = f"hub.foundries.io/{args.factory}/"
+
+    # We don't need to validate the compose content here - just need to
+    # iterate service images.
+    ComposeApps.App.DockerComposeTool = None
+
+    apps = ComposeApps(args.apps_root)
+    analyzed = {}
+    for app in apps:
+        for img in app.images():
+            img = expandvars(img)
```

Comment:
Maybe it makes sense to move it to `ComposeApps.Apps.images`.

Reply:
https://github.com/foundriesio/ci-scripts/pull/208/commits/892c7a21f8571d4d1b523dda05f7d88502787468